Update Golang "Advanced Topics" page to reference operator-lib

## Bug Report

<!-- 
Note: Make sure to first check the prerequisites that can be found in the main README file!

Thanks for filing an issue! Before hitting the button, please answer these questions.
Fill in as much of the template below as you can. If you leave out information, we can't help you as well.
-->

**What did you do?**
Create a new operator using v0.19.2 and add status.Conditions as documented in [Advanced Topics](https://sdk.operatorframework.io/docs/golang/advanced-topics/)

**What did you expect to see?**
`make test` reports same output as before

```bash
go: creating new go.mod: module tmp
go: found sigs.k8s.io/controller-tools/cmd/controller-gen in sigs.k8s.io/controller-tools v0.3.0
/home/jlfowle/go/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
go fmt ./...
go vet ./...
/home/jlfowle/go/bin/controller-gen "crd:trivialVersions=true" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
go test ./... -coverprofile cover.out
?       github.com/jlfowle/sonarqube-operator   [no test files]
?       github.com/jlfowle/sonarqube-operator/api/v1alpha1      [no test files]
ok      github.com/jlfowle/sonarqube-operator/controllers       3.811s  coverage: 0.0% of statements
```

**What did you see instead? Under which circumstances?**

When `replace k8s.io/client-go => k8s.io/client-go v0.18.2` is not included.

```bash
go: creating new go.mod: module tmp
go: found sigs.k8s.io/controller-tools/cmd/controller-gen in sigs.k8s.io/controller-tools v0.3.0
/home/jlfowle/go/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
go fmt ./...
go vet ./...
# k8s.io/client-go/tools/clientcmd/api/v1
/home/jlfowle/go/pkg/mod/k8s.io/client-go@v12.0.0+incompatible/tools/clientcmd/api/v1/conversion.go:29:15: scheme.AddConversionFuncs undefined (type *runtime.Scheme has no field or method AddConversionFuncs)
/home/jlfowle/go/pkg/mod/k8s.io/client-go@v12.0.0+incompatible/tools/clientcmd/api/v1/conversion.go:31:12: s.DefaultConvert undefined (type conversion.Scope has no field or method DefaultConvert)
/home/jlfowle/go/pkg/mod/k8s.io/client-go@v12.0.0+incompatible/tools/clientcmd/api/v1/conversion.go:34:12: s.DefaultConvert undefined (type conversion.Scope has no field or method DefaultConvert)
/home/jlfowle/go/pkg/mod/k8s.io/client-go@v12.0.0+incompatible/tools/clientcmd/api/v1/conversion.go:37:12: s.DefaultConvert undefined (type conversion.Scope has no field or method DefaultConvert)
/home/jlfowle/go/pkg/mod/k8s.io/client-go@v12.0.0+incompatible/tools/clientcmd/api/v1/conversion.go:40:12: s.DefaultConvert undefined (type conversion.Scope has no field or method DefaultConvert)
/home/jlfowle/go/pkg/mod/k8s.io/client-go@v12.0.0+incompatible/tools/clientcmd/api/v1/conversion.go:43:12: s.DefaultConvert undefined (type conversion.Scope has no field or method DefaultConvert)
/home/jlfowle/go/pkg/mod/k8s.io/client-go@v12.0.0+incompatible/tools/clientcmd/api/v1/conversion.go:46:12: s.DefaultConvert undefined (type conversion.Scope has no field or method DefaultConvert)
/home/jlfowle/go/pkg/mod/k8s.io/client-go@v12.0.0+incompatible/tools/clientcmd/api/v1/conversion.go:49:12: s.DefaultConvert undefined (type conversion.Scope has no field or method DefaultConvert)
/home/jlfowle/go/pkg/mod/k8s.io/client-go@v12.0.0+incompatible/tools/clientcmd/api/v1/conversion.go:52:12: s.DefaultConvert undefined (type conversion.Scope has no field or method DefaultConvert)
# sigs.k8s.io/controller-runtime/pkg/metrics
/home/jlfowle/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/metrics/client_go_adapter.go:133:24: not enough arguments in call to metrics.Register
/home/jlfowle/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/metrics/client_go_adapter.go:133:25: undefined: metrics.RegisterOpts
/home/jlfowle/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/metrics/workqueue.go:99:48: cannot use workqueueMetricsProvider literal (type workqueueMetricsProvider) as type workqueue.MetricsProvider in argument to workqueue.SetProvider:
        workqueueMetricsProvider does not implement workqueue.MetricsProvider (missing NewDeprecatedAddsMetric method)
# sigs.k8s.io/controller-runtime/pkg/client
/home/jlfowle/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/client/typed_client.go:46:5: too many arguments in call to o.resourceMeta.Interface.Post().NamespaceIfScoped(o.Object.GetNamespace(), o.resourceMeta.isNamespaced()).Resource(o.resourceMeta.resource()).Body(obj).VersionedParams(createOpts.AsCreateOptions(), c.paramCodec).Do
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/client/typed_client.go:65:5: too many arguments in call to o.resourceMeta.Interface.Put().NamespaceIfScoped(o.Object.GetNamespace(), o.resourceMeta.isNamespaced()).Resource(o.resourceMeta.resource()).Name(o.Object.GetName()).Body(obj).VersionedParams(updateOpts.AsUpdateOptions(), c.paramCodec).Do
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/client/typed_client.go:84:5: too many arguments in call to o.resourceMeta.Interface.Delete().NamespaceIfScoped(o.Object.GetNamespace(), o.resourceMeta.isNamespaced()).Resource(o.resourceMeta.resource()).Name(o.Object.GetName()).Body(deleteOpts.AsDeleteOptions()).Do
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/client/typed_client.go:103:5: too many arguments in call to o.resourceMeta.Interface.Delete().NamespaceIfScoped(deleteAllOfOpts.ListOptions.Namespace, o.resourceMeta.isNamespaced()).Resource(o.resourceMeta.resource()).VersionedParams(deleteAllOfOpts.ListOptions.AsListOptions(), c.paramCodec).Body(deleteAllOfOpts.DeleteOptions.AsDeleteOptions()).Do
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/client/typed_client.go:126:5: too many arguments in call to o.resourceMeta.Interface.Patch(patch.Type()).NamespaceIfScoped(o.Object.GetNamespace(), o.resourceMeta.isNamespaced()).Resource(o.resourceMeta.resource()).Name(o.Object.GetName()).VersionedParams(patchOpts.ApplyOptions(opts).AsPatchOptions(), c.paramCodec).Body(data).Do
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/client/typed_client.go:139:20: too many arguments in call to r.Interface.Get().NamespaceIfScoped(key.Namespace, r.isNamespaced()).Resource(r.resource()).Name(key.Name).Do
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/client/typed_client.go:154:5: too many arguments in call to r.Interface.Get().NamespaceIfScoped(listOpts.Namespace, r.isNamespaced()).Resource(r.resource()).VersionedParams(listOpts.AsListOptions(), c.paramCodec).Do
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/client/typed_client.go:175:5: too many arguments in call to o.resourceMeta.Interface.Put().NamespaceIfScoped(o.Object.GetNamespace(), o.resourceMeta.isNamespaced()).Resource(o.resourceMeta.resource()).Name(o.Object.GetName()).SubResource("status").Body(obj).VersionedParams((&UpdateOptions literal).ApplyOptions(opts).AsUpdateOptions(), c.paramCodec).Do
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/client/typed_client.go:199:5: too many arguments in call to o.resourceMeta.Interface.Patch(patch.Type()).NamespaceIfScoped(o.Object.GetNamespace(), o.resourceMeta.isNamespaced()).Resource(o.resourceMeta.resource()).Name(o.Object.GetName()).SubResource("status").Body(data).VersionedParams(patchOpts.ApplyOptions(opts).AsPatchOptions(), c.paramCodec).Do
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/client/unstructured_client.go:56:5: too many arguments in call to o.resourceMeta.Interface.Post().NamespaceIfScoped(o.Object.GetNamespace(), o.resourceMeta.isNamespaced()).Resource(o.resourceMeta.resource()).Body(obj).VersionedParams(createOpts.AsCreateOptions(), uc.paramCodec).Do
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/client/unstructured_client.go:56:5: too many errors
# k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1beta1
/home/jlfowle/go/pkg/mod/k8s.io/apiextensions-apiserver@v0.18.4/pkg/client/clientset/clientset/typed/apiextensions/v1beta1/customresourcedefinition.go:72:5: too many arguments in call to c.client.Get().Resource("customresourcedefinitions").Name(name).VersionedParams(&options, scheme.ParameterCodec).Do
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/k8s.io/apiextensions-apiserver@v0.18.4/pkg/client/clientset/clientset/typed/apiextensions/v1beta1/customresourcedefinition.go:88:5: too many arguments in call to c.client.Get().Resource("customresourcedefinitions").VersionedParams(&opts, scheme.ParameterCodec).Timeout(timeout).Do
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/k8s.io/apiextensions-apiserver@v0.18.4/pkg/client/clientset/clientset/typed/apiextensions/v1beta1/customresourcedefinition.go:104:8: too many arguments in call to c.client.Get().Resource("customresourcedefinitions").VersionedParams(&opts, scheme.ParameterCodec).Timeout(timeout).Watch
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/k8s.io/apiextensions-apiserver@v0.18.4/pkg/client/clientset/clientset/typed/apiextensions/v1beta1/customresourcedefinition.go:114:5: too many arguments in call to c.client.Post().Resource("customresourcedefinitions").VersionedParams(&opts, scheme.ParameterCodec).Body(customResourceDefinition).Do
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/k8s.io/apiextensions-apiserver@v0.18.4/pkg/client/clientset/clientset/typed/apiextensions/v1beta1/customresourcedefinition.go:127:5: too many arguments in call to c.client.Put().Resource("customresourcedefinitions").Name(customResourceDefinition.ObjectMeta.Name).VersionedParams(&opts, scheme.ParameterCodec).Body(customResourceDefinition).Do
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/k8s.io/apiextensions-apiserver@v0.18.4/pkg/client/clientset/clientset/typed/apiextensions/v1beta1/customresourcedefinition.go:142:5: too many arguments in call to c.client.Put().Resource("customresourcedefinitions").Name(customResourceDefinition.ObjectMeta.Name).SubResource("status").VersionedParams(&opts, scheme.ParameterCodec).Body(customResourceDefinition).Do
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/k8s.io/apiextensions-apiserver@v0.18.4/pkg/client/clientset/clientset/typed/apiextensions/v1beta1/customresourcedefinition.go:153:5: too many arguments in call to c.client.Delete().Resource("customresourcedefinitions").Name(name).Body(&opts).Do
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/k8s.io/apiextensions-apiserver@v0.18.4/pkg/client/clientset/clientset/typed/apiextensions/v1beta1/customresourcedefinition.go:168:5: too many arguments in call to c.client.Delete().Resource("customresourcedefinitions").VersionedParams(&listOpts, scheme.ParameterCodec).Timeout(timeout).Body(&opts).Do
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/k8s.io/apiextensions-apiserver@v0.18.4/pkg/client/clientset/clientset/typed/apiextensions/v1beta1/customresourcedefinition.go:181:5: too many arguments in call to c.client.Patch(pt).Resource("customresourcedefinitions").Name(name).SubResource(subresources...).VersionedParams(&opts, scheme.ParameterCodec).Body(data).Do
        have (context.Context)
        want ()
# k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1
/home/jlfowle/go/pkg/mod/k8s.io/apiextensions-apiserver@v0.18.4/pkg/client/clientset/clientset/typed/apiextensions/v1/customresourcedefinition.go:72:5: too many arguments in call to c.client.Get().Resource("customresourcedefinitions").Name(name).VersionedParams(&options, scheme.ParameterCodec).Do
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/k8s.io/apiextensions-apiserver@v0.18.4/pkg/client/clientset/clientset/typed/apiextensions/v1/customresourcedefinition.go:88:5: too many arguments in call to c.client.Get().Resource("customresourcedefinitions").VersionedParams(&opts, scheme.ParameterCodec).Timeout(timeout).Do
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/k8s.io/apiextensions-apiserver@v0.18.4/pkg/client/clientset/clientset/typed/apiextensions/v1/customresourcedefinition.go:104:8: too many arguments in call to c.client.Get().Resource("customresourcedefinitions").VersionedParams(&opts, scheme.ParameterCodec).Timeout(timeout).Watch
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/k8s.io/apiextensions-apiserver@v0.18.4/pkg/client/clientset/clientset/typed/apiextensions/v1/customresourcedefinition.go:114:5: too many arguments in call to c.client.Post().Resource("customresourcedefinitions").VersionedParams(&opts, scheme.ParameterCodec).Body(customResourceDefinition).Do
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/k8s.io/apiextensions-apiserver@v0.18.4/pkg/client/clientset/clientset/typed/apiextensions/v1/customresourcedefinition.go:127:5: too many arguments in call to c.client.Put().Resource("customresourcedefinitions").Name(customResourceDefinition.ObjectMeta.Name).VersionedParams(&opts, scheme.ParameterCodec).Body(customResourceDefinition).Do
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/k8s.io/apiextensions-apiserver@v0.18.4/pkg/client/clientset/clientset/typed/apiextensions/v1/customresourcedefinition.go:142:5: too many arguments in call to c.client.Put().Resource("customresourcedefinitions").Name(customResourceDefinition.ObjectMeta.Name).SubResource("status").VersionedParams(&opts, scheme.ParameterCodec).Body(customResourceDefinition).Do
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/k8s.io/apiextensions-apiserver@v0.18.4/pkg/client/clientset/clientset/typed/apiextensions/v1/customresourcedefinition.go:153:5: too many arguments in call to c.client.Delete().Resource("customresourcedefinitions").Name(name).Body(&opts).Do
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/k8s.io/apiextensions-apiserver@v0.18.4/pkg/client/clientset/clientset/typed/apiextensions/v1/customresourcedefinition.go:168:5: too many arguments in call to c.client.Delete().Resource("customresourcedefinitions").VersionedParams(&listOpts, scheme.ParameterCodec).Timeout(timeout).Body(&opts).Do
        have (context.Context)
        want ()
/home/jlfowle/go/pkg/mod/k8s.io/apiextensions-apiserver@v0.18.4/pkg/client/clientset/clientset/typed/apiextensions/v1/customresourcedefinition.go:181:5: too many arguments in call to c.client.Patch(pt).Resource("customresourcedefinitions").Name(name).SubResource(subresources...).VersionedParams(&opts, scheme.ParameterCodec).Body(data).Do
        have (context.Context)
        want ()
make: *** [Makefile:63: vet] Error 2
```


**Environment**
* operator-sdk version:

v0.19.2

* go version:

go version go1.14.6 linux/amd64

* Kubernetes version information:

Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.6", GitCommit:"dff82dc0de47299ab66c83c626e08b245ab19037", GitTreeState:"clean", BuildDate:"2020-07-15T16:58:53Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.3", GitCommit:"012b3ec", GitTreeState:"clean", BuildDate:"2020-07-22T21:49:20Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}

* Kubernetes cluster kind: 

openshift

* Are you writing your operator in ansible, helm, or go?

**Possible Solution**

Add to instructions for including replacement of `k8s.io/client-go` version in [Advanced Topics](https://sdk.operatorframework.io/docs/golang/advanced-topics/)

**Additional context**
Add any other context about the problem here.

--- a/website/content/en/docs/building-operators/golang/advanced-topics.md
+++ b/website/content/en/docs/building-operators/golang/advanced-topics.md
@@ -19,7 +19,7 @@ To use conditions in your custom resource, add a Conditions field to the Status 
 
 ```Go
 import (
-    "github.com/operator-framework/operator-sdk/pkg/status"
+    "github.com/operator-framework/operator-lib/status"
 )
 
 type MyAppStatus struct {

--- a/website/content/en/docs/building-operators/golang/advanced-topics.md
+++ b/website/content/en/docs/building-operators/golang/advanced-topics.md
@@ -247,7 +247,7 @@ A call to `leader.Become()` will block the operator as it retries until it can b
 ```Go
 import (
     ...
-    "github.com/operator-framework/operator-sdk/pkg/leader"
+    "github.com/operator-framework/operator-lib/leader"
 )
 
 func main() {

